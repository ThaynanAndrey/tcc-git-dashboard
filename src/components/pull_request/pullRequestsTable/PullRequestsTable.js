import React from 'react';
import PullRequestsTableHead from './PullRequestsTableHead';
import PullRequestsTableBody from './PullRequestsTableBody';
import NewPullRequestsTableBody from './NewPullRequestsTableBody';

/**
 * Stateless component with the function of presenting a table of Pull Requests.
 * 
 * @author Thaynan Nunes
 */
const PullRequestsTable = ({ pullRequests, isListNewPullRequests, addPullRequest, removePullRequest, openPullRequest }) => {
    const pullRequestTableBody = isListNewPullRequests
        ? <NewPullRequestsTableBody pullRequests={pullRequests} addPullRequest={addPullRequest}/>
        : <PullRequestsTableBody pullRequests={pullRequests} removePullRequest={removePullRequest}
            openPullRequest={openPullRequest}/>;

    return (
        <table className="striped highlight responsive-table">
            <PullRequestsTableHead />
            {pullRequestTableBody}
        </table>
    )
}

export default PullRequestsTable;