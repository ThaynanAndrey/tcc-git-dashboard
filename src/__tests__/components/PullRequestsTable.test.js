import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestsTable from '../../components/pull_request/pullRequestsTable/PullRequestsTable';

describe('Test PullRequestsTable', () => {
  it('renders correctly a PullRequestsTable\'s component with correct props', () => {
    const pullRequests = [ { name: "Tests", dataCriacao: "10/10/1010", status: "open",
        responsavel: { nome: "Ze" }, repositorio: { nome: "Repo" } } ];
    const [addPullRequest, removePullRequest, openPullRequest] = Array(3).fill(jest.fn());
    const isListNewPullRequests = true;

    const tree = mount(
      <PullRequestsTable
        pullRequests={pullRequests}
        isListNewPullRequests={isListNewPullRequests}
        addPullRequest={addPullRequest}
        removePullRequest={removePullRequest}
        openPullRequest={openPullRequest}
      />
    );

    expect(toJson(tree)).toMatchSnapshot();

    expect(tree.prop('pullRequests')).toEqual(pullRequests);
    expect(tree.prop('isListNewPullRequests')).toEqual(isListNewPullRequests);
    expect(tree.prop('addPullRequest')).toEqual(addPullRequest);
    expect(tree.prop('removePullRequest')).toEqual(removePullRequest);
    expect(tree.prop('openPullRequest')).toEqual(openPullRequest);
  });

  it('renders correctly a PullRequestsTable\'s component with correct classes', () => {
    const pullRequests = [ { name: "Tests", dataCriacao: "10/10/1010", status: "open",
        responsavel: { nome: "Ze" }, repositorio: { nome: "Repo" } } ];
    const [addPullRequest, removePullRequest, openPullRequest] = Array(3).fill(jest.fn());
    const isListNewPullRequests = false;

    const tree = mount(
      <PullRequestsTable
        pullRequests={pullRequests}
        isListNewPullRequests={isListNewPullRequests}
        addPullRequest={addPullRequest}
        removePullRequest={removePullRequest}
        openPullRequest={openPullRequest}
      />
    );

    expect(toJson(tree)).toMatchSnapshot();

    expect(tree.prop('pullRequests')).toEqual(pullRequests);
    expect(tree.prop('isListNewPullRequests')).toEqual(isListNewPullRequests);
    expect(tree.prop('addPullRequest')).toEqual(addPullRequest);
    expect(tree.prop('removePullRequest')).toEqual(removePullRequest);
    expect(tree.prop('openPullRequest')).toEqual(openPullRequest);
  });
});