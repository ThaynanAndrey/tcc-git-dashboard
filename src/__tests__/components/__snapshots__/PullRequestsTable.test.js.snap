// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestsTable renders correctly a PullRequestsTable's component with correct classes 1`] = `
<PullRequestsTable
  addPullRequest={[MockFunction]}
  isListNewPullRequests={false}
  openPullRequest={[MockFunction]}
  pullRequests={
    Array [
      Object {
        "dataCriacao": "10/10/1010",
        "name": "Tests",
        "repositorio": Object {
          "nome": "Repo",
        },
        "responsavel": Object {
          "nome": "Ze",
        },
        "status": "open",
      },
    ]
  }
  removePullRequest={[MockFunction]}
>
  <table
    className="striped highlight responsive-table"
  >
    <PullRequestsTableHead>
      <thead>
        <tr>
          <th>
            Pull Request
          </th>
          <th>
            Data
          </th>
          <th>
            Status
          </th>
          <th>
            Respons치vel
          </th>
          <th>
            Reposit칩rio
          </th>
          <th />
        </tr>
      </thead>
    </PullRequestsTableHead>
    <PullRequestsTableBody
      openPullRequest={[MockFunction]}
      pullRequests={
        Array [
          Object {
            "dataCriacao": "10/10/1010",
            "name": "Tests",
            "repositorio": Object {
              "nome": "Repo",
            },
            "responsavel": Object {
              "nome": "Ze",
            },
            "status": "open",
          },
        ]
      }
      removePullRequest={[MockFunction]}
    >
      <tbody>
        <tr
          key="0"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <td />
          <td>
            10/10/1010
          </td>
          <td>
            open
          </td>
          <td>
            Ze
          </td>
          <td>
            Repo
          </td>
          <td>
            <Tooltip.Wrapper
              arrow={8}
              background="#000"
              border="#000"
              color="#fff"
              content="Excluir PR"
              fadeDuration={0}
              fadeEasing="linear"
              fixed={false}
              fontFamily="inherit"
              fontSize="inherit"
              offset={0}
              padding={16}
              placement="left"
              radius={10}
              style={
                Object {
                  "whiteSpace": "nowrap",
                }
              }
              zIndex={1}
            >
              <EmotionCssPropInternal
                __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                css={
                  Object {
                    "map": undefined,
                    "name": "1oin9j2",
                    "next": undefined,
                    "styles": "
          
  position: relative;
  display: inline-block;
;
          
        ",
                  }
                }
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "whiteSpace": "nowrap",
                  }
                }
              >
                <div
                  className="css-1oin9j2"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <i
                    className="material-icons left red-text"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    delete
                  </i>
                  <Tooltip
                    fadeDuration={0}
                    fadeEasing="linear"
                    offset={8}
                    open={false}
                    placement="left"
                    zIndex={1}
                  />
                </div>
              </EmotionCssPropInternal>
            </Tooltip.Wrapper>
          </td>
        </tr>
      </tbody>
    </PullRequestsTableBody>
  </table>
</PullRequestsTable>
`;

exports[`Test PullRequestsTable renders correctly a PullRequestsTable's component with correct props 1`] = `
<PullRequestsTable
  addPullRequest={[MockFunction]}
  isListNewPullRequests={true}
  openPullRequest={[MockFunction]}
  pullRequests={
    Array [
      Object {
        "dataCriacao": "10/10/1010",
        "name": "Tests",
        "repositorio": Object {
          "nome": "Repo",
        },
        "responsavel": Object {
          "nome": "Ze",
        },
        "status": "open",
      },
    ]
  }
  removePullRequest={[MockFunction]}
>
  <table
    className="striped highlight responsive-table"
  >
    <PullRequestsTableHead>
      <thead>
        <tr>
          <th>
            Pull Request
          </th>
          <th>
            Data
          </th>
          <th>
            Status
          </th>
          <th>
            Respons치vel
          </th>
          <th>
            Reposit칩rio
          </th>
          <th />
        </tr>
      </thead>
    </PullRequestsTableHead>
    <NewPullRequestsTableBody
      addPullRequest={[MockFunction]}
      pullRequests={
        Array [
          Object {
            "dataCriacao": "10/10/1010",
            "name": "Tests",
            "repositorio": Object {
              "nome": "Repo",
            },
            "responsavel": Object {
              "nome": "Ze",
            },
            "status": "open",
          },
        ]
      }
    >
      <tbody>
        <tr
          key="0"
        >
          <td />
          <td>
            10/10/1010
          </td>
          <td>
            open
          </td>
          <td>
            Ze
          </td>
          <td>
            Repo
          </td>
          <td>
            <Tooltip.Wrapper
              arrow={8}
              background="#000"
              border="#000"
              color="#fff"
              content="Adicionar PR"
              fadeDuration={0}
              fadeEasing="linear"
              fixed={false}
              fontFamily="inherit"
              fontSize="inherit"
              offset={0}
              padding={16}
              placement="left"
              radius={10}
              style={
                Object {
                  "whiteSpace": "nowrap",
                }
              }
              zIndex={1}
            >
              <EmotionCssPropInternal
                __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
                css={
                  Object {
                    "map": undefined,
                    "name": "1oin9j2",
                    "next": undefined,
                    "styles": "
          
  position: relative;
  display: inline-block;
;
          
        ",
                  }
                }
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "whiteSpace": "nowrap",
                  }
                }
              >
                <div
                  className="css-1oin9j2"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <i
                    className="material-icons left green-text"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    add_box
                  </i>
                  <Tooltip
                    fadeDuration={0}
                    fadeEasing="linear"
                    offset={8}
                    open={false}
                    placement="left"
                    zIndex={1}
                  />
                </div>
              </EmotionCssPropInternal>
            </Tooltip.Wrapper>
          </td>
        </tr>
      </tbody>
    </NewPullRequestsTableBody>
  </table>
</PullRequestsTable>
`;
